### ISSUE TITLE
Not all parameters are configured, can't execute. Please run the setup steps.

### ISSUE THREAD

**Issue by Nahar09** [SharePoint/sp-dev-modernization#48]
On clicking 'Create Modern Version' button on a web part page , receiving the following error.
Not all parameters are configured, can't execute. Please run the setup steps.
AzureAD application ID:
AzureAD function host:
Page transformation endpoint:
However , if I rerun the setup , it says APP is already installed.

**Comment by jansenbe** [comment]
Hi @Nahar09 ,

Please use PnP PowerShell to run `Get-PnPStorageEntity` and provide me with the output of the entries starting with "Modernization_". There should be 6 entries.

**Comment by yhertel** [comment]
I'm having the same issue.  I ran the above and here is my output.  
![modernizationscreenshot1](https://user-images.githubusercontent.com/46942794/52133931-a5334680-25ff-11e9-9048-053af1198e24.jpg)

**Comment by jansenbe** [comment]
@yhertel, that looks good. Can you check the following (copied from #46)

This looks like the bug I've fixed yesterday. When you've enabled page transformation for a site did you use the same casing for the URL as the URL actually is shown in SharePoint. There's a string replacement that used to fail when not being case sensitive.

Two ways to fix (both should work)

Drop the latest solution from dev tree (https://github.com/SharePoint/sp-dev-modernization/blob/dev/Solutions/PageTransformationUI/assets/sharepointpnp-pagetransformation-central.sppkg) in the app catalog and then go to the modernization center -> site contents -> click on the three dots next to the sharepointpnp-pagetransformation-central app and click on "Get it" to install the latest version in the modernization center site
Use the web part in the modernization center and disable page transformation for your site and then re-enable it using the url written in the correct casing

**Comment by yhertel** [comment]
Do I need to blow away and re-create the Modernization center site?   I did both of the fixes listed above and still have the problem.  For me, if I try to run it in Edge, I get the 'Not all parameters configured...' but if I run in IE or Chrome, I get the 'Busy generating a modern version...' forever and ever.   It also doesn't appear to add the toolbar button to sites -- I thought I'd try with a fresh site that I hadn't used it on before, and even though the Modernization center says it installed it, it doesn't appear in the ribbon.

**Comment by jansenbe** [comment]
When you press F12 (developer console) what errors do you see in the console?

**Comment by jansenbe** [comment]
Did you do the admin consent for the created Azure AD app? When you did the Azure setup the script mentioned at the very end to do the admin consent via a URL like this "Open a browser session to https://login.microsoftonline.com/common/oauth2/authorize?client_id=f0e040f0-21e3-4640-ba50-7b56be765b26&response_type=code&prompt=admin_consent"

So in your case, the Azure AD App id for you is the 6773811d-... value shown from the earlier `Get-PnPStorageEntity` call. So can you construct the needed URL (https://login.microsoftonline.com/common/oauth2/authorize?client_id=6773811d-TOCOMPLETE&response_type=code&prompt=admin_consent) and run that as that will get the consent done. 

Once that's done please ensure that you enable the Page Transformation using the exact URL casing as SharePoint is showing when you browse the site (as mentioned earlier in this thread).

**Comment by yhertel** [comment]
I am relying on another group to configure the Azure side because I do not have access -- makes it very challenging.   I will forward to them and hopefully this will do the trick!  Thank you for your assistance!

**Comment by jansenbe** [comment]
Hi @yhertel ,

Wanted to make you aware of the troubleshooting guide I created: https://github.com/SharePoint/sp-dev-modernization/blob/master/Solutions/PageTransformationUI/docs/troubleshootingguide.md.

Would be good if you could check the SharePoint side and have the Azure team verify the Azure side as well.

**Comment by jansenbe** [comment]
Hi @Nahar09,

Please be aware that there's a troubleshooting guide that can hopefully unblock you (if you still see the issue at least): https://github.com/SharePoint/sp-dev-modernization/blob/master/Solutions/PageTransformationUI/docs/troubleshootingguide.md

**Comment by Nahar09** [comment]
Also for me, similar to @yhertel  , the problem is in edge only , if i run it in Chrome or IE , I get the 'Busy generating a modern version...' and nothing happens.

**Comment by jansenbe** [comment]
Hi @Nahar09,

Can you also provide the output from `Get-PnPTenantServicePrincipalPermissionGrants` + when it's in the "Busy generating a modern version..." state press F12, go to the console output and create a screenshot so I can see the possible errors.

Also the troubleshooting guide (https://github.com/SharePoint/sp-dev-modernization/blob/master/Solutions/PageTransformationUI/docs/troubleshootingguide.md) can be helpful.

**Comment by Nahar09** [comment]
here is the output
![image](https://user-images.githubusercontent.com/47241354/52262734-34549d00-2925-11e9-984d-0d696c8b5cd1.png)
![image](https://user-images.githubusercontent.com/47241354/52262832-7aa9fc00-2925-11e9-9b8d-79c4d2133e7c.png)

**Comment by Nahar09** [comment]
The console gives following errors
![image](https://user-images.githubusercontent.com/47241354/52263116-3703c200-2926-11e9-86a5-c5d1246f4f1a.png)
![image](https://user-images.githubusercontent.com/47241354/52263150-4b47bf00-2926-11e9-88ae-6145caec0b3f.png)
![image](https://user-images.githubusercontent.com/47241354/52263190-69adba80-2926-11e9-8a5c-c472769f702f.png)

**Comment by jansenbe** [comment]
Hey @Nahar09,

The problem is that you are pointing to a non existing Azure function app...The **Modernization_FunctionHost** points to pnpmodernizationtest1.azurewebsites.net but if I try to do a DNS lookup I don't find that host address. Did you do the Azure setup part? Could it be that the Azure function app was deleted afterwards?

**Comment by jansenbe** [comment]
Hi @Nahar09,

Did the above help explain the issue you see?

**Comment by jansenbe** [comment]
@yhertel : how are things progressing?

**Comment by yhertel** [comment]
@jansenbe I'm still at the error re: the SPO Client Extensibility Web Application Principal.  The Azure Admin says he did it all, but this doesn't appear anywhere, so it's obviously not set up correctly.  I read somewhere that it can sometimes be accidentally deleted?  But the AzureAdmin says he can't access the Azure powershell to restore it.  After being an on-prem admin where I had access to everything I needed, I'm really struggling with SPO and having to rely on other groups to troubleshoot.  I've been instructed to not spend any more time on it.  Thank you for your efforts on this.

**Comment by jansenbe** [comment]
@yhertel : thx for the update. The SPO Client Extensibility Web Application Principal is a SharePoint infrastructure app and should be created automatically whenever the first deployed application needs it. It might be that since you do not have permissions in the AD side that this cause a problem. A solution would be to have a global admin do the deployment.

@Nahar09 : I've not heard back from you, feel free to create a new issue with problems you see after you've checked the missing Azure Function app setup

**Comment by Nahar09** [comment]
Thanks for all the help. Will work on this and try to resolve the issue. Will also check the troubleshooting guide.
